src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +144,1336 @@ static DamageTable D_80BBF858 = {
+            // Previously facing player, but now player out of range.
+            // Start counter for frames/function calls to ignore player while in process of returning to idle.
+            this->loseAttentionTimer = 20;
+        }
+        // Start animation to return to idle position.
+        this->actionFlags &= ~EN_S_GORO_ACTIONFLAG_FACEPLAYER;
+        this->headRotZ = 0;
+        this->headRotY = 0;
+        this->bodyRotZ = 0;
+        this->bodyRotY = 0;
+    }
+    return true;
+}
+
+void EnSGoro_SetupAction(EnSGoro* this, PlayState* play) {
+    if (Object_IsLoaded(&play->objectCtx, this->loadedObjIndex)) {
+        this->actionFlags = 0;
+        if (EN_S_GORO_OFTYPE_WSHRINE) {
+            if (gSaveContext.save.weekEventReg[22] & 4) {
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_GKQUIET_ACKNOWLEDGED;
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_EYESOPEN;
+                this->animInfoIndex = EN_S_GORO_ANIM_SHIVER_A;
+                SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+            } else {
+                this->eyeTexIndex = EN_S_GORO_EYETEX_CLOSED2;
+                this->actionFlags |= EN_S_GORO_ACTIONFLAG_EARSCOVERED;
+                this->animInfoIndex = EN_S_GORO_ANIM_COVEREARS;
+                SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+            }
+        } else {
+            this->actionFlags |= EN_S_GORO_ACTIONFLAG_EYESOPEN;
+            this->animInfoIndex = EN_S_GORO_ANIM_IDLE_STAND;
+            SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+        }
+
+        this->scaleFactor = 0.01f;
+        Actor_SetScale(&this->actor, 0.01f);
+        this->actor.gravity = -1.0f;
+        this->actor.flags |= ACTOR_FLAG_10;
+        this->actor.flags |= ACTOR_FLAG_2000000;
+        this->actor.targetMode = 1;
+
+        switch (EN_S_GORO_GET_MAIN_TYPE(&this->actor)) {
+            case EN_S_GORO_TYPE_SHRINE_WINTER_A:
+                this->actionFunc = EnSGoro_WinterShrineGoron_Idle;
+                return;
+            case EN_S_GORO_TYPE_SHRINE_WINTER_B:
+                this->actionFunc = EnSGoro_WinterShrineGoron_Idle;
+                return;
+            case EN_S_GORO_TYPE_SHRINE_WINTER_C:
+                this->actionFunc = EnSGoro_WinterShrineGoron_Idle;
+                return;
+            case EN_S_GORO_TYPE_SHRINE_SPRING_A:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_SpringShrineGoron_Idle;
+                    return;
+                }
+                return;
+            case EN_S_GORO_TYPE_SHRINE_SPRING_B:
+                if (EnSGoro_SetPartnerActor_GoronElder(this, play)) {
+                    this->actionFunc = EnSGoro_SpringShrineGoron_Idle;
+                    return;
+                }
+                return;
```

Comment:
```suggestion
                }
                break;
```
When you have two returns like this (or when they're changed to breaks), you really don't need the first one. Only the second one. Applies to a lot in this function